docs/graphcast/radios/intro.md

```diff
@@ -31,251 +28,196 @@ Once that's done, you can start building your very first Radio.
-:::
+Open your `main.rs` file and add the following imports:
 
-`logger` will be used to structure logs in a better way than the good ol' `console.log` (very important if we want to have better tracing and metrics in the future).
+```Rust
+use ethers::{
+    providers::{Http, Middleware, Provider},
+    types::U64,
+};
+use graphcast_sdk::gossip_agent::message_typing::GraphcastMessage;
+use graphcast_sdk::gossip_agent::GossipAgent;
+use once_cell::sync::OnceCell;
+use std::env;:
+use std::sync::{Arc, Mutex};
+use std::{thread::sleep, time::Duration};
+use tokio::sync::Mutex as AsyncMutex;
+```
 
 ### Structure
 
-The most basic structure of our ping pong code that we can have is to put everything below the imports and the constants in one `run()` function, which can looks something like this:
+Everything we need will be inside the `main()` function. And since we'll be using async code we have to annotate it with `#[tokio::main]`, we can start off with something as simple as:
 
-```typescript
-const run = async () => {
-  // TODO: Ping pong 🏓
-};
-
-run()
-  .then()
-  .catch((err) => {
-    logger.error(`❌ Oh no! An error occurred: ${err.message}`);
-    process.exit(1);
-  });
 ```
-
-From here on out everything we write will reside in the `run()` function.
+#[tokio::main]
+async fn main() {
+  // TODO: Radio logic
+}
+```
 
 ### Instantiate the essentials
 
-Let's instantiate a few classes that will do all the heavy lifting for us:
+Let's instantiate a few variables that will do all the heavy lifting for us.
 
-```typescript
-const clientManager = new ClientManager({
-  operatorPrivateKey: process.env.RADIO_OPERATOR_PRIVATE_KEY,
-  ethNodeUrl: process.env.ETH_NODE,
-  registry: process.env.REGISTRY_SUBGRAPH,
-  graphNodeStatus: process.env.GRAPH_NODE,
-  indexerManagementServer: process.env.INDEXER_MANAGEMENT_SERVER,
-  graphNetworkUrl: process.env.NETWORK_SUBGRAPH,
-});
+First is the vector we'll use to store incoming messages:
 
-const gossipAgent = new GossipAgent(logger, clientManager);
+```Rust
+pub static MESSAGES: OnceCell<Arc<Mutex<Vec<GraphcastMessage>>>> = OnceCell::new();
 ```
 
-:::info
-Don't worry about the environment variables that you see there which we haven't added to the `.env`, they will reside in the `Dockerfile` that we'll create at the end.
-:::
+That line may look intimidating, but really it's just defining a global static instance of a `GraphcastMessage` vector. It is used to save incoming messages after they've been validated, in order to defer their processing for later, because async code is required for the processing but is not allowed inside the handler function.
```

Comment:
I'm not sure if we should explain the details here. For easier understanding maybe we try to use positive languages? something like 
`The vector keeps track of validated messages from the gossip operator, allowing the radios to freely process the messages separately`?